### PR TITLE
Add business status on BE

### PR DESCRIPTION
**Pull request by jbendzala**

### CODE REVIEW

File: lobis-repo/server/prisma/schema.prisma [hunk 1]
```diff
@@ -41,12 +41,13 @@ model DeliveryDetail {
...
 
 model Business {
-  id              String     @id @default(uuid())
+  id              String          @id @default(uuid())
   name            String
   description     String?
   images          String[]
   belongsToUserId String
   categories      Category[]
+  businessStatus  BusinessStatus?
```
Comment (cypcz) [inline lobis-repo/server/prisma/schema.prisma:50]: I'd name it just `status` (the enum name can stay as is). It's obvious that it's business status if it's on business entity.

File: lobis-repo/server/src/handlers/admin/business/business.ts [hunk 2]
```diff
@@ -55,6 +55,7 @@ export const createBusiness = async (req: Request, res: Response) => {
             images,
             categories,
             belongsToUserId: user.id,
+            businessStatus: 'PENDING',
```
Comment (cypcz) [inline lobis-repo/server/src/handlers/admin/business/business.ts:58]: use the enum here :)

File: lobis-repo/server/src/handlers/admin/business/business.ts [hunk 3]
```diff
@@ -66,7 +67,7 @@ export const updateBusiness = async (req: Request, res: Response) => {
 
     assert(req.body, updateBusinessValidator)
 
-    const { name, description, images, categories } = req.body
+    const { name, description, images, categories, businessStatus } = req.body
```
Comment (cypcz) [inline lobis-repo/server/src/handlers/admin/business/business.ts:70]: why is the status in the input? user should not be able to change it..

File: lobis-repo/server/src/handlers/public/business/business.ts [hunk 4]
```diff
@@ -12,6 +12,23 @@ export const getBusinesses = async (req: Request, res: Response) => {
     const businesses = await prisma.business.findMany({
         where: {
             deletedAt: null,
+            businessStatus: 'ACTIVE',
```
Comment (cypcz) [inline lobis-repo/server/src/handlers/public/business/business.ts:15]: use enum

**Review by cypcz — CHANGES_REQUESTED** [review]

File: lobis-repo/server/prisma/schema.prisma (hunk 1 above)
Comment (jbendzala) [reply lobis-repo/server/prisma/schema.prisma:50]: Good point, renamed

File: lobis-repo/server/src/routes/public/business.ts [hunk 5]
```diff
@@ -1,9 +1,14 @@
...
+import {
+    getBusiness,
+    getBusinesses,
+    getPendingBusinesses,
+} from '../../handlers/public/business/business'
 
 const router = Router({ mergeParams: true })
 
 router.get('/', getBusinesses)
+router.get('/pending', getPendingBusinesses)
```
[the remark this answers is not in the document]
Comment (jbendzala) [reply lobis-repo/server/src/routes/public/business.ts:11]: This is exactly what I was aiming for. In the future, there will be some "Approver" or "Super admin" or something like that, with access to this  hidden "approving" part of the app. Should I leave it like it is, or change it to query filter?

File: lobis-repo/server/src/handlers/admin/business/business.ts (hunk 3 above)
Comment (jbendzala) [reply lobis-repo/server/src/handlers/admin/business/business.ts:70]: In the future, I'd like to change statuses from some hidden "approving" part of the app. Should I create a different endpoint only for status change then?

**Pull request merged**